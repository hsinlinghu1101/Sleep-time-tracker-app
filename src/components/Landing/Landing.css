h3{
    color:#ffe867;
}
.login, .register{
    padding: 0 3%;
    margin: 1% 3%;
    border-style:solid;
    display: inline-block;
    
}
