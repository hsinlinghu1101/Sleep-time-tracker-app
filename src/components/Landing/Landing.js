import React from 'react'
import { Link } from 'react-router-dom'
import sleep1 from '../../image/sleep1.jpg';
import sleep2 from '../../image/sleep2.jpg'
import './Landing.css'

export default function Landing(){
    return(
    <main role="main">
      <section>
        <img className='sleep one' src={sleep1} alt="sleep-left"/>
        <p className='empty'></p>
        <img  className='sleep two' src={sleep2} alt="sleep-right"/>
      </section>
      <section>
        <div>
            <h3>Introduction</h3>
        </div>
        
        <p>In todayâ€™s fast-paced society, six or seven hours of sleep may sound pretty good. In reality, the average adult sleeps less than seven hours per night.Sleep Time Tracker record your total hours of sleep </p>
      </section>
      <section>
        <div>
            <h3>Track your sleeping hours</h3>
        </div>
        <p>Sleep Time Tracker is built for people above 14 years old who want to record daily total hours of sleep. You can input the time when you go to bed and when you wake up. Sleep Time Tracker will display the result and provide the feedback based on your age. </p>
      </section>
      
      <section>
        <p>Let's track your sleeping hours</p>
        <div className='register'><Link to='/register'>Register</Link></div>
        <div className='login'><Link to='/login'>Login</Link></div>
      </section>
    </main>
  
    )
}