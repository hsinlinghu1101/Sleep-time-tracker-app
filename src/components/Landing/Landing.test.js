import React from 'react';
import ReactDOM from 'react-dom';
import {MemoryRouter as Router} from 'react-router-dom';
import Landing from './Landing';
import renderer from 'react-test-renderer';

it('renders without crashing', ()=>{
    const div = document.createElement('div')

    ReactDOM.render(
        <Router>
        <Landing />
        </Router>, div)
        ReactDOM.unmountComponentAtNode(div)
    
})
it('renders the UI as expected', () => {
    const tree = renderer
      .create(
        <Router>
          <Landing />
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });