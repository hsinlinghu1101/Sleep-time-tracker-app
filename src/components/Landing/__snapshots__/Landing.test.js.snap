// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<main
  role="main"
>
  <section
    className="landingImg"
  >
    <img
      alt="sleep"
      className="sleep"
      src="sleep.jpg"
    />
  </section>
  <section>
    <div>
      <h3>
        Introduction
      </h3>
    </div>
    <p>
      In todayâ€™s fast-paced society, six or seven hours of sleep may sound pretty good. In reapty, the average adult sleeps less than seven hours per night. Sleep Time Tracker records your total hours of sleep per day.
    </p>
  </section>
  <section>
    <div>
      <h3>
        Track your sleeping hours
      </h3>
    </div>
    <p>
      Sleep Time Tracker was built for people above 14 years old who want to keep track of daily total hours of sleep. You can input the time when you went to bed and when you woke up. Sleep Time Tracker will display the result and provide feedback based on your age.
    </p>
  </section>
  <section>
    <div>
      <h3>
        How to use it
      </h3>
    </div>
    <div
      className="instruction"
    >
      <p>
        Step 1 - If you are a new user, please register an account. If you already have an account please login directly.
      </p>
      <p>
        Step 2 - After login, you can start to record your sleep hour.
      </p>
      <p>
        Step 3 - After submitting the form, you can find all your records.
      </p>
      <p>
        Step 4 - You can delete or add record when you log in.
      </p>
      <div
        className="demo"
      >
        <p>
          If you'd like to try this app, log in with our demo user account 
        </p>
        <p>
          username: Demo_user
        </p>
        <p>
          password: Demouserpass123!
        </p>
        <p>
           This person is between 18 - 64 years old
        </p>
      </div>
    </div>
  </section>
  <section>
    <h3>
      Let's track your sleeping hours
    </h3>
    <div
      className="register"
    >
      <a
        href="/register"
        onClick={[Function]}
      >
        Register
      </a>
    </div>
    <div
      className="login"
    >
      <a
        href="/login"
        onClick={[Function]}
      >
        Login
      </a>
    </div>
  </section>
  <div
    className="end"
  />
</main>
`;
